src/schemaToPropTypes.js

```diff
@@ -0,0 +1,195 @@
+ */
+const getIndentation = (indentLevel = INDENT_LEVEL) =>
+	indentLevel && typeof indentLevel === 'number'
+		? [...Array(Math.trunc(indentLevel))].map(() => `${INDENT_CHAR}`).join('')
+		: '';
+
+/**
+ * Decorator to add indentation to the strings output from any needed functions.
+ * @param {Function} func - The function returning a string whose return needs to be decorated.
+ * @param {Boolean} prefix - Whether the indentation should happen as a prefix or as a suffix. Prefix is default
+ * @returns {function(args: Array): string} - The decorated String
+ * @private
+ */
+const _indentDecorator = (func, prefix = true) => args => {
+	const indents = getIndentation();
+
+	return `${prefix ? indents : ''}${func.apply(this, args)}${prefix ? '' : indents}`;
+};
+
+/**
+ * Adds the `Required` string given a property and an array of required properties.
+ * @param {String} propName - The name of the property to be checked.
+ * @param {Object} property - The property itself (is property.required a valid openAPI format?).
+ * @param {Array} requiredProps - The array of required properties in a schema.
+ * @returns {string}
+ */
+const getRequired = (propName, property, requiredProps = []) =>
+	Array.isArray(requiredProps) && (requiredProps.indexOf(propName) > -1 || property.required)
+		? '.isRequired,'
+		: ',';
+
+/**
+ * Returns the name of the reference.
+ * @param {String} ref - The `$ref` string.
+ * @returns {string} - The name of the reference in the `$ref`
+ */
+const getRef = ref => ref.split('/').pop();
+
+/**
+ * Creates the string for the value of a property.
+ * @param {String} propertyName - The key (name) of the property.
+ * @param {Object} property - The property to generate the value from.
+ * @returns {string}
+ */
+const getPropTypeValue = (propertyName, property) => {
+	let str = 'PropTypes.';
+
+	switch (property.type) {
+		case 'array':
+			if (property.items.$ref) {
+				const extractRefProp = formatComponentName(getRef(property.items.$ref));
+				str += `arrayOf(${extractRefProp})`;
+			} else {
+				str += `arrayOf(${getPropTypeValue(propertyName, property.items)})`;
+			}
+			break;
+
+		case 'object':
+			if (property.$ref) {
+				str += `shape(${formatComponentName(getRef(property.$ref))})`;
+			} else {
+				INDENT_LEVEL += 1;
+				const indentation = getIndentation(INDENT_LEVEL - 1);
+				// eslint-disable-next-line no-use-before-define
+				str += `shape({\n${getPropTypes(propertyName, property)}${indentation}})`;
+				INDENT_LEVEL -= 1;
+			}
+			break;
+
+		case 'number':
+		case 'integer':
+		case 'long':
+		case 'float':
+		case 'double':
+			str += 'number';
+			break;
+
+		case 'string':
+		case 'byte':
+		case 'binary':
+		case 'date':
+		case 'DATETIME':
+		case 'password':
+			str += 'string';
+			break;
+
+		case 'boolean':
+			str += 'bool';
+			break;
+
+		default:
+			if (property.$ref) {
+				str += getPropTypeValue(propertyName, { type: 'object', ...property });
```

Comment:
Noticed one thing, if we have in OpenAPI just red it goes into `default` case here and runs `getPropTypeValue` with `type `object` set, but this causes double `PropTypes.` quick workaround was to just put here `str =` instead of `str += ` but I'm not sure if that's the best.
Any bette solution for this?